### PR TITLE
FEAT: Implement redux and modals controller

### PR DESCRIPTION
**Pull request by gor017**
### What are you trying to accomplish?
[BF-3]
[BF-12]

### CODE REVIEW

File: src/pages/profile/applications/index.tsx [hunk 1]
```diff
@@ -41,73 +42,71 @@ const applications = [
...
+          <Button
+            variant="text"
+            leftIcon={<DeleteIcon />}
+            onClick={() =>
+              dispatch(
+                viewsMiddleware.openModal({
+                  name: ModalName.DeleteApplicationModal,
+                  props: {},
+                })
+              )
```
Comment (ArthurO2002) [inline src/pages/profile/applications/index.tsx:84]: Move to function

File: src/pages/profile/applications/index.tsx [hunk 2]
```diff
@@ -41,73 +42,71 @@ const applications = [
...
+      size="hg"
+      leftIcon={<AddIcon />}
+      onClick={() =>
+        dispatch(
+          viewsMiddleware.openModal({
+            name: ModalName.AddApplicationModal,
+            props: {},
+          })
+        )
+      }
```
Comment (ArthurO2002) [inline src/pages/profile/applications/index.tsx:104]: move

**Review by ArthurO2002 — CHANGES_REQUESTED** [review]

File: src/pages/profile/applications/index.tsx (hunk 1 above)
Comment (gor017) [reply src/pages/profile/applications/index.tsx]: Done

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -40,15 +43,18 @@ const App = ({ Component, pageProps }: IAppProps) => {
...
-            <Component {...pageProps} />
-          </ChildLayout>
-        ) : (
-          <Layout>
-            <Component {...pageProps} />
-          </Layout>
-        )}
+        <Provider store={store}>
+          {ChildLayout ? (
+            <ChildLayout>
```
Comment (Antonyan7) [inline src/pages/_app.tsx:48]: HierarchicalLayout

**Review by Antonyan7 — APPROVED** [review]

File: src/components/Modals/Application/AddApplicationModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import { CloseIcon, JobIcon } from '@components/Icons'
+import { Dialog } from '@headlessui/react'
+import { dispatch } from '@redux/hooks'
+import { viewsMiddleware } from '@redux/slices/views'
+import { Roboto } from 'next/font/google'
+
+const roboto = Roboto({
+  weight: ['400', '500'],
+  subsets: ['latin'],
+  variable: '--font-roboto',
```
[the remark this answers is not in the document]
Comment (rikarsen) [reply src/components/Modals/Application/AddApplicationModal/index.tsx:13]: also this

**Review by rikarsen — CHANGES_REQUESTED** [review]

File: src/components/Modals/Application/AddApplicationModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+})
+
+export const AddApplicationModal = () => {
+  const onClose = useCallback(() => {
+    dispatch(viewsMiddleware.closeModal(ModalName.AddApplicationModal))
+  }, [])
+
+  return (
+    <Dialog
```
[the remark this answers is not in the document]
Comment (gor017) [reply src/components/Modals/Application/AddApplicationModal/index.tsx:22]: https://beparmenia.atlassian.net/jira/software/c/projects/BF/boards/1?modal=detail&selectedIssue=BF-23&assignee=61b21390ef18ca0071f96582

File: package.json [hunk 6]
```diff
@@ -15,18 +15,22 @@
...
     "@headlessui/react": "^1.7.13",
+    "@reduxjs/toolkit": "^1.9.3",
     "@storybook/addon-essentials": "7.0.0-rc.4",
     "@storybook/addon-interactions": "7.0.0-rc.4",
     "@storybook/addon-links": "7.0.0-rc.4",
     "@storybook/blocks": "7.0.0-rc.4",
     "@storybook/nextjs": "7.0.0-rc.4",
     "@storybook/react": "7.0.0-rc.4",
     "@storybook/testing-library": "0.0.14-next.1",
+    "@types/react-redux": "^7.1.25",
```
[the remark this answers is not in the document]
Comment (gor017) [reply package.json]: Done

File: src/pages/_app.tsx (hunk 3 above)
Comment (gor017) [reply src/pages/_app.tsx]: Done

File: src/components/Modals/Application/AddApplicationModal/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (gor017) [reply src/components/Modals/Application/AddApplicationModal/index.tsx:23]: Done

File: src/components/Modals/Application/AddApplicationModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import React, { useCallback } from 'react'
+import { ModalName } from '@allTypes/modals'
+import { Button } from '@components/Button'
+import { CloseIcon, JobIcon } from '@components/Icons'
+import { Dialog } from '@headlessui/react'
+import { dispatch } from '@redux/hooks'
+import { viewsMiddleware } from '@redux/slices/views'
+import { Roboto } from 'next/font/google'
+
+const roboto = Roboto({
```
[the remark this answers is not in the document]
Comment (rikarsen) [reply src/components/Modals/Application/AddApplicationModal/index.tsx:10]: https://beparmenia.atlassian.net/browse/BF-29

**Review by rikarsen — APPROVED** [review]

**Pull request merged**